### PR TITLE
Ensure tests the modify same git repo don't execute at the same time

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
A subset of the rapids_cpm tests modify the same git repo active branch which isn't safe. We need to serialize those tests to ensure everything works robustly.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: testing/utils/cmake_test.cmake [hunk 1]
```diff
@@ -141,6 +142,17 @@ function(add_cmake_test mode source_or_dir)
       message(FATAL_ERROR "${mode} mode not one of the valid modes (config|build|install) by add_cmake_build_test")
     endif()
 
+    if(RAPIDS_TEST_SERIAL)
+      set_tests_properties(${test_name} PROPERTIES RUN_SERIAL ON)
+      if(TEST ${test_name}_configure)
+        message(STATUS "${test_name}_configure is serial")
```
Comment (vyasr) [inline testing/utils/cmake_test.cmake:148]: Slightly inconsistent with a status message for the configure but not the build.

**Review by vyasr — APPROVED** [review]
LGTM (one minor comment). Thanks!

File: testing/utils/cmake_test.cmake (hunk 1 above)
Comment (robertmaynard) [reply testing/utils/cmake_test.cmake:148]: Ohh that was a debug line, will remove it.

**Pull request merged**